src/read.c

```diff
@@ -4009,15 +4009,15 @@ static avifResult avifParse(avifDecoder * decoder)
             if (readResult != AVIF_RESULT_OK) {
                 return readResult;
             }
-            if (header.size == 0) {
+            if (header.isSizeZeroBox) {
                 header.size = boxContents.size;
             } else if (boxContents.size != header.size) {
                 // A truncated box, bail out
                 return AVIF_RESULT_TRUNCATED_DATA;
             }
         } else if (header.size > (UINT64_MAX - parseOffset)) {
             return AVIF_RESULT_BMFF_PARSE_FAILED;
-        } else if (header.size == 0) {
+        } else if (header.isSizeZeroBox) {
```

Comment:
I suggest reversing the checks at line 4018 and line 4020. We should use `header.size` only when `header.isSizeZeroBox` is false.

Reply:
Done.